### PR TITLE
[QC-210] Custom Data Sampling Condtions using ROOT reflection

### PR DESCRIPTION
**Pull request by knopers8**
@Barthelemy

### CODE REVIEW

File: Framework/Core/src/DataSamplingConditionCustom.cxx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    // it does not seem to be documented anywhere, but this pointer to
+    // a dictionary should not be deleted - it results in segfaults.
+    TClass* dictionary = TClass::GetClass(className.c_str());
+    if (!dictionary) {
+      throw std::runtime_error("Failed to load the dictionary of the class: " + className + " from the library: " + libraryName);
+    }
+
+    mCondition.reset(static_cast<DataSamplingCondition*>(dictionary->New()));
```
Comment (Barthelemy) [inline Framework/Core/src/DataSamplingConditionCustom.cxx]: can dictionary->New() fail ? what happens if it fails ?

Comment (knopers8) [reply Framework/Core/src/DataSamplingConditionCustom.cxx]: It should print an error message, invoke an error handler (default or globally defined) and return 0. If the handler throws, we should catch the exception, if it aborts - I don't think there is anything we can do.

Comment (Barthelemy) [inline Framework/Core/src/DataSamplingConditionCustom.cxx]: What happens if dictionary->New() fails ?

**Comment by knopers8** [comment]
Every required build is passing, I propose merging before something breaks again in CI.

**Pull request merged**